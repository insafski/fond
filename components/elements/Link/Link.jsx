import React from "react";
import NextLink from "next/link";
import { useRouter } from "next/router";
import PropTypes from "prop-types";
import get from "lodash/get";

export default function Link({ children, href, locale }) {
	const router = useRouter();

	return (
		<NextLink href={`${href}`} locale={locale ? locale : get(router, "locale", "")}>
			{children}
		</NextLink>
	);
}

Link.propTypes = {
	children: PropTypes.node,
	href: PropTypes.string,
	locale: PropTypes.string,
};

Link.defaultProps = {
	children: null,
	href: "/",
	locale: "",
};
