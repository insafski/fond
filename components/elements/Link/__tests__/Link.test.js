import React from "react";
import renderer from "react-test-renderer";

import Link from "..";

describe("Link", () => {
	it("renders correctly", () => {
		const tree = renderer
			.create(<Link href={"/"}>
				Ссылка на страницу
			</Link>)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
