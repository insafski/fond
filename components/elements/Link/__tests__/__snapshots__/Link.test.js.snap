// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Ссылка на страницу
</a>
`;
